plugins/deployment_manager/DeploymentManager.ts

```diff
@@ -240,9 +247,31 @@ export class DeploymentManager {
     return await this.cache.readCache(getArtifactSpec(migration));
   }
 
-  async clone<C extends Contract>(address: string, args: any[], network?: string): Promise<C> {
+  /**
+   * Call an async function with a given amount of retries
+   * @param fn an async function that takes a signer as an argument. The function takes a signer
```

Comment:
> a new instance of a signer needs to be used on each retry

how come?

also, `this.getSigner()` will return the same instance of the signer, right?

Reply:
oh, i see; you're resetting `this._signers` to force it to generate new instances

Reply:
An ethers signer wraps around a provider, and we need to use a new provider to submit a retry txn when the old txn is hanging. Otherwise, the retry txn will just hang again using the old provider.